WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -392,12 +392,35 @@ def time_until_competition
     }
   end
 
-  def currency_convert
-    update_rates_if_needed
-    converted = Money.new(params[:value], params[:from_currency]).exchange_to(params[:to_currency])
+  def calculate_dues
+    country_iso2 = Country.where(id: params[:country_id]).pluck(:iso2)[0]
+    country_band = CountryBand.where(iso2: country_iso2).map(&:number)[0]
+
+    update_currency_rates_if_needed
+    currency_subunit = Money::Currency.new(params[:currency_code]).subunit_to_unit
+    money_value = params[:entry_fee].to_i / currency_subunit
+    money_value_usd = Money.new(money_value, params[:currency_code]).exchange_to("USD").cents
```

Comment:
There is `Money.from_amount(human_entered_amount, currency_iso)` which completely saves you from reading the currency and parsing the subunit above.

Reply:
I don't see how this function would help...we still need to divide `params[:entry_fee]` by the subunit to unit ratio because, in USD for example, there is always two zeroes attached behind every amount entered (because we read it from HTML). If the user entered 20 then `params[:entry_fee]` would be 2000 and `from_amount(2000, "USD")` would consider the input to be 2000 dollars and not 20 dollars. Maybe `Money` has a function like `from_cents` that would make everything easier?  No luck finding it for now...

Reply:
Guess what, the constructor takes in the fractional unit, so Money.new(entered_amount, currency).exchange_to("USD") is all we need! lol